### PR TITLE
LPS-98033 Changes suggested in @wincent PR

### PR DESCRIPTION
**Pull request by boton**

### CODE REVIEW

File: modules/apps/flags/flags-taglib/src/main/resources/META-INF/resources/flags/react/js/components/FlagsModal.es.js [hunk 1]
```diff
@@ -54,15 +54,15 @@ const ModalContentForm = ({
...
 			<div className="form-group">
-				<label className="control-label" htmlFor="reason">
+				<label className="control-label" htmlFor="selectedReason">
 					{Liferay.Language.get('reason-for-the-report')}
 				</label>
 				<select
 					className="form-control"
-					id="reason"
-					name="reason"
+					id="selectedReason"
```
Comment (wincent) [inline modules/apps/flags/flags-taglib/src/main/resources/META-INF/resources/flags/react/js/components/FlagsModal.es.js]: Sorry I didn't catch this the first time, but we need to watch out for id clashes (although there may only be one "flags modal" on the page at a time, "selectedReason" might conceivably collide at some point in the future). Can we pick something with a bit more "uniqueness"?

File: modules/apps/flags/flags-taglib/src/main/resources/META-INF/resources/flags/react/js/index.es.js [hunk 2]
```diff
@@ -18,7 +18,7 @@ import ReactDOM from 'react-dom';
 import Flags from './components/Flags.es';
 import ThemeContext from './ThemeContext.es';
 
-export default function(id, props, context) {
+export default function reacDomRenderFlags(id, props, context) {
```
Comment (wincent) [inline modules/apps/flags/flags-taglib/src/main/resources/META-INF/resources/flags/react/js/index.es.js]: Typo: "reac" -> "react"

**Comment by jbalsas** [comment]
I've resent this to myself for a final pass at https://github.com/jbalsas/liferay-portal/pull/1845. Will continue over there

**Comment by jbalsas** [comment]
ci:close

**Pull request closed without merging**